### PR TITLE
Use metadata to handle the reference relationship between kvstore and dict

### PR DESCRIPTION
**Pull request by soloestoy**
Feature `one-dict-per-slot` refactors the database, and part of it involved splitting the rehashing list from the global level back to the database level, or more specifically, the kvstore level. This change is fine, and it also simplifies the process of swapping databases, which is good. And it should not have a major impact on the efficiency of incremental rehashing.

To implement the kvstore-level rehashing list, each `dict` under the `kvstore` needs to know which `kvstore` it belongs. However, kvstore did not insert the reference relationship into the `dict` itself, instead, it placed it in the `dictType`. In my view, this is a somewhat odd way. Theoretically, `dictType` is just a collection of function handles, a kind of virtual type that can be referenced globally, not an entity. But now the `dictType` is instantiated, with each `kvstore` owning an actual `dictType`, which in turn holds a reverse reference to the `kvstore`'s resource pointer. This design is somewhat uncomfortable for me.

I think the `dictType` should not be instantiated. The references between actual resources (`kvstore` and `dict`) should occur between specific objects, rather than force materializing the `dictType`, which is supposed to be virtual.

### CODE REVIEW

File: src/server.c [hunk 1]
```diff
@@ -474,26 +474,32 @@ dictType zsetDictType = {
     NULL,              /* allow to expand */
 };
 
-/* Db->dict, keys are sds strings, vals are Objects. */
-dictType dbDictType = {
+/* Kvstore->keys, keys are sds strings, vals are Objects. */
+dictType kvstoreDictType = {
```
Comment (zuiderkwast) [inline src/server.c]: Include "Db" to distinguish from the other kvstore dict types.

```suggestion
dictType kvstoreDbDictType = {
```

**Review by zuiderkwast — COMMENTED** [review]
Thanks! I had the feeling that something is a bad design here.

File: src/server.c (hunk 1 above)
Comment (hpatro) [reply src/server.c]: The wording we finalized for a `db` were `keys` and `expires`. So, this can be `kvstoreKeysDictType`.

**Review by hpatro — APPROVED** [review]

**Comment by soloestoy** [comment]
The test failed again. Has anyone else encountered this issue, or is it also a timing issue? @enjoy-binbin needs your help : )

**Comment by enjoy-binbin** [comment]
seem like a timing issue, it is an old issue, i can take a look later, we can ignore it for now.

link: https://github.com/valkey-io/valkey/actions/runs/10627531881/job/29460790842?pr=966

**Comment by zuiderkwast** [comment]
I opened #969 to track the test failure.

I think we can merge this PR, but we need to focus on the flaky tests before the 8.0 release so everything is more stable. There is a label [test-failure](https://github.com/valkey-io/valkey/issues?q=is%3Aissue+is%3Aopen+label%3Atest-failure) for these tests.

**Review by zuiderkwast — APPROVED** [review]

**Review by lilpoozie2005 — APPROVED** [review]

**Pull request merged**